File: 528-swapping-nodes-in-a-linked-list/swapping-nodes-in-a-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def swapNodes(self, head: Optional[ListNode], k: int) -> Optional[ListNode]:
        p, q = head, head
        l = 0
        while p:
            l += 1
            p = p.next
        p = head
        for _ in range(k - 1):
            p = p.next
        for _ in range(l - k):
            q = q.next
        temp = p.val
        p.val = q.val
        q.val = temp
        return head

File: 528-swapping-nodes-in-a-linked-list/swapping-nodes-in-a-linked-list.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
struct ListNode* swapNodes(struct ListNode* head, int k) {
    struct ListNode* p = head;
    struct ListNode* q = head;

    int len=0, temp;
    for (;p;p=p->next)len++;

    p = head;
    for (int i = 0; i < k-1; i++)p=p->next;
    for (int i = 0; i < len-k; i++)q=q->next;

    temp=p->val;
    p->val=q->val;
    q->val=temp;
    return head;
}